mkdocs/docs/HPC/setting_up_python_virtual_environments.md

```diff
@@ -0,0 +1,262 @@
+$ module load PyTorch/2.1.2-foss-2023a
+$ python -m venv myenv
+$ source myenv/bin/activate
+# install virtual environment dependencies
+$ pip install Poutyne
+$ deactivate
+```
+
+We exit the interactive shell by pressing `CTRL+D` and create a job script that loads the PyTorch module, 
+enters the virtual environment and executes the script:
+
+```bash title="jobscript.pbs"
+#!/bin/bash
+
+# Basic parameters
+#PBS -N python_job_example    ## Job name
+#PBS -l nodes=1:ppn=1         ## 1 node, 1 processors per node
+#PBS -l walltime=01:00:00     ## Max time your job will run (no more than 72:00:00)
+
+module load PyTorch/2.1.2-foss-2023a   # Load the PyTorch module
+
+cd $PBS_O_WORKDIR             # Change working directory to the location where the job was submitted
+
+source myenv/bin/activate     # Activate the virtual environment
+python pytorch_poutyne.py     # Run your Python script, or any other command within the virtual environment
+deactivate                    # Deactivate the virtual environment
+```
+
+Next, we submit the job script:
+
+```bash
+$ qsub jobscript.pbs
+```
+
+Two files will be created in the directory where the job was submitted: `python_job_example.o[job_id]` and `python_job_example.e[job_id]`.
+The `.o` file contains the output of the job.
+
+
+## Troubleshooting
+
+### Illegal instruction error
+
+Activating a virtual environment created on a different cluster can cause issues. 
+This happens
+because the binaries in the virtual environments from cluster A might not work with the CPU architecture of cluster B.
+
+For example, if we create a virtual environment on the skitty cluster,
+
+```bash
+$ module swap cluster/skitty
+$ qsub -I
+$ python -m venv myenv
+```
+
+return to the login node by pressing CTRL+D and try to use the virtual environment:
+
+```bash
+$ source myenv/bin/activate
+$ python
+Illegal instruction (core dumped)
+```
+
+we are presented with the illegal instruction error. More info on this [here](troubleshooting.md#illegal-instruction-error)
+
+### Error: cannot open shared object file: No such file or directory
+
+There are two main reasons why this error could occur.
+
+1. you have not loaded the python module that was used to create the virtual environment.
+2. you added or removed modules while the virtual environment was activated.
+
+#### Entering a virtual environment while the python module used to create it is not active
+
+When you load a python module and use that to make a virtual environment, you need to make sure that the same module 
+is loaded when you enter the environment. This is because the virtual environment keeps a reference to the base python 
+used to create it.
+
+The following commands illustrate this issue:
+
+```bash
+$ module load Python/3.10.8-GCCcore-12.2.0  # Load a python module
+$ python -m venv myenv                      # Create a virtual environment with loaded python module
+$ module purge                              # Remove all loaded modules
```

Comment:
mention `(WRONG!)` or something in the comment to highlight that this is where things go wrong, and maybe use `module unload Python` instead to make it a bit more explicit?

Mentioning `module purge` is good too though, since that may be more common than selective unloading modules

Reply:
Fixed in [clarify wrong command in example](https://github.com/hpcugent/vsc_user_docs/pull/639/commits/1a2f3f1a2cb4041e239bbaad9a3b454d6b084c13)